docs/reference/snapshot-restore/apis/restore-snapshot-api.asciidoc

```diff
@@ -0,0 +1,230 @@
+[[restore-snapshot-api]]
+=== Restore snapshot API
+++++
+<titleabbrev>Restore snapshot</titleabbrev>
+++++
+
+Restores a <<snapshot-restore,snapshot>> of a cluster or specified data streams and indices.
+
+////
+[source,console]
+----
+PUT /_snapshot/my_repository
+{
+  "type": "fs",
+  "settings": {
+    "location": "my_backup_location"
+  }
+}
+
+PUT /_snapshot/my_repository/my_snapshot?wait_for_completion=true
+
+PUT /index_1
+
+PUT /index_2
+
+PUT /index_3
+
+PUT /index_4
+
+PUT _snapshot/my_repository/snapshot_2?wait_for_completion=true
+{
+  "indices": "index_3,index_4",
+  "ignore_unavailable": true,
+  "include_global_state": false,
+  "metadata": {
+    "taken_by": "Elastic Machine",
+    "taken_because": "backup testing"
+  }
+}
+
+POST /index_1/_close
+
+POST /index_2/_close
+
+POST /index_3/_close
+
+POST /index_4/_close
+
+----
+// TESTSETUP
+////
+
+[source,console]
+----
+POST /_snapshot/my_repository/my_snapshot/_restore
+----
+// TEST[s/_restore/_restore?wait_for_completion=true/]
+
+[[restore-snapshot-api-request]]
+==== {api-request-title}
+
+`POST /_snapshot/<repository>/<snapshot>`
+
+[[restore-snapshot-api-desc]]
+==== {api-description-title}
+
+Use the restore snapshot API to restore a snapshot of a cluster, including all data streams and indices in the snapshot. If you do not want to restore the entire snapshot, you can select specific data streams or indices to restore.
+
+NOTE: You cannot restore a data stream if a stream with the same name already
+exists.
+
+You can run the restore operation only on a functioning cluster. Existing indices can only be restored if they are <<indices-close,closed>> and have the same number of shards as the indices in the snapshot. The restore operation automatically opens restored indices if they were closed and creates new indices if they do not exist in the cluster.
```

Comment:
> You can run the restore operation only on a functioning cluster.

`functioning` is a little vague here. Does this mean the cluster is running? Green health status? Both?

Reply:
I believe that it's any _running_ cluster. @original-brownbear, is there also a limitation on the cluster health?

Reply:
The health of the cluster is not relevant but you do need to at least have an elected master node as well as enough data nodes to handle what you are planning on restoring. I think I'd just drop the `functioning` from this sentence and be done with it. If you don't have a running master then you don't really have a cluster I guess :)

Reply:
Thanks for the clarification @original-brownbear. I think that adding some more precise language here is helpful. 
>You can run the restore operation on a cluster that contains an elected
primary node and has data nodes with enough capacity to accommodate the snapshot
you are restoring.

Reply:
@lockewritesdocs that's ok as well, just make it `an elected master node` instead of `primary` and it's correct :)

Reply:
@original-brownbear, as part of the effort to introduce neutral naming, we're trying to provide alternatives to terms like `master` and `slave`. That's why I suggested `elected primary node` instead of `elected master node`.

Reply:
I think we can't adjust terminology here yet :( Especially not to "primary" as we use "primary" in the context of primary and replica shards. We'll have to adjust naming here once we renamed the actual piece of functionality I'm afraid.

Reply:
I agree with @original-brownbear.

Here, `master` node refers to an actual setting for the node in ES:
https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-node.html#dedicated-master-node

I don't feel we can change this or other remaining master node references until that setting is changed. Otherwise, it could be very confusing for users.

FWIW I feel renaming this is important and it's being worked on, but the product needs to make this change first.